developer_guides/debugability/coredump-reader/index.rst

```diff
@@ -30,3 +30,47 @@ We read from dump file into sof-coredump-reader.py, then we pipe its output to x
 .. code-block:: bash
 
    ./sof-coredump-to-gdb.sh sof-apl dump_file
+
+Usage with Linux SOF Driver
+***************************
+
+Upon DSP errors where a coredump is generated, the Linux SOF
+driver allows to access the dump via debugfs. An example how
+to capture the dump file and process it with coredump-reader:
+
+.. code-block:: bash
+
+   dut> cat /sys/kernel/debug/sof/exception >dsp-coredump
+   # transfer file to host
+   host> sof/tools/coredumper/sof-coredump-reader.py  -v -l 4 -i dsp-coredump -o dsp-coredump.gdb
+   host> xt-gdb sof/build_tlg_xcc/sof --command=dsp-coredump.gdb
+   [cut]
+   $1 = "Exception location:"
+   0xbe02fb29 is in ipc_glb_debug_message (/home/user/sof/src/ipc/handler-ipc3.c:1371).
+   [cut]
+   $2 = "backtrace"
+   #0  0xbe051b00 in literals ()
+   #1  0xbe04e277 in dump_stack (p=3187705884, addr=0x1cc6c29b, offset=3270769662, limit=380, stack_ptr=0x1) at /home/user//sof/src/arch/xtensa/include/arch/lib/cache.h:79
+   #2  0xbe04e2f7 in panic_dump (p=233492486, panic_info=0x0, data=0xbe0a4130) at /home/user/sof/src/arch/xtensa/include/arch/debug/panic.h:45
+   #3  0xbe02dfd9 in exception () at /home/user/sof/src/arch/xtensa/init.c:115
+   #4  0xbe050a28 in _GeneralException ()
+   #5  0xbe02fb29 in ipc_glb_debug_message (header=394016) at /home/user/sof/src/ipc/handler-ipc3.c:1373
+   [cut]
+   (xt-gdb) info all-registers
+   pc             0xbe051b00       0xbe051b00 <literals>
+   ar0            0x0      0
+   ar1            0xbe00a044       -1107255228
+   ar2            0x10000  65536
+
+Notes:
+ - coredump reader only works with xcc toolchain
```

Comment:
Is this simply because `xt-gdb` is used? Or is there another issue if gcc or clang is used?

Reply:
@cujomalainey It's primarily due to xt-gdb. That should be "just work" to fix in the coredumper script, but we've also had some weird issues with gcc built binaries. See e.g. this old bug https://github.com/thesofproject/sof/issues/1346 .

Reply:
yeah, so gcc has a slightly different calling ABI AFAICT vs XCC and the stack trace will look wrong. Regular gdb also interprets this incorrectly (but otherwise works at runtime debug on qemu). Maybe something for Maciej to look at in the future.